src/annotator/anchoring/range-js.js

```diff
@@ -32,3 +34,51 @@ export function nodeFromXPath(xpath, root) {
   });
   return node;
 }
+
+/**
+ * Attempt to evaluate a provided xpath. If the evaluation fails, then fall back to a
+ * manual evaluation using `nodeFromXPathFallback` that can construct a simple xpath
+ * that is somewhat reasonable, albeit perhaps incomplete or less accurate.
+ */
+function evaluateXPath(xp, root, nsResolver = null) {
+  try {
+    return document.evaluate(
+      '.' + xp,
+      root,
+      nsResolver,
+      XPathResult.FIRST_ORDERED_NODE_TYPE,
+      null
+    ).singleNodeValue;
+  } catch (e) {
+    // There are cases when the evaluation fails, because the
+    // HTML documents contains nodes with invalid names,
+    // for example tags with equal signs in them, or something like that.
+    // In these cases, the XPath expressions will have these abominations,
+    // too, and then they can not be evaluated.
+    // In these cases, we get an XPathException, with error code 52.
+    // See http://www.w3.org/TR/DOM-Level-3-XPath/xpath.html#XPathException
+    // This does not necessarily make any sense, but this what we see
+    // happening.
+    // eslint-disable-next-line no-console
+    console.log('XPath evaluation failed.');
+    // eslint-disable-next-line no-console
+    console.log('Trying fallback...');
+    // We have a an 'evaluator' for the really simple expressions that
+    // should work for the simple expressions we generate.
+    return nodeFromXPathFallback(xp, root);
+  }
+}
+
+/**
+ * Finds an element node using an xpath relative to the document root.
+ *
+ * Example:
+ *   node = nodeFromXPath('/html/body/div/p[2]')
+ */
+export function nodeFromXPath(xpath, root = document) {
+  if (!$.isXMLDoc(document.documentElement)) {
+    return evaluateXPath(xpath, root);
+  } else {
+    throw new Error(`XML documents are not supported. xpath: ${xpath}`);
```

Comment:
I think you could simplify this by just removing the `$.isXMLDoc` check and always attempting to use the `evaluateXPath` code path. If it fails, higher level logic will skip to trying one of the other selectors.